pkg/auth/auth.go

```diff
@@ -118,10 +119,11 @@ func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginO
 	}
 
 	// If no username and no password is specified, try to use existing ones.
+	serverDomain := domain(server)
 	if opts.Username == "" && password == "" && authConfig.Username != "" && authConfig.Password != "" {
-		fmt.Println("Authenticating with existing credentials...")
-		if err := docker.CheckAuth(ctx, systemContext, authConfig.Username, authConfig.Password, server); err == nil {
-			fmt.Fprintln(opts.Stdout, "Existing credentials are valid. Already logged in to", server)
+		fmt.Fprintln(opts.Stdout, "Authenticating with existing credentials for", server)
+		if err := docker.CheckAuth(ctx, systemContext, authConfig.Username, authConfig.Password, serverDomain); err == nil {
```

Comment:
Shouldn’t this check the relevant repo, not just the general server? But then again, if there were credentials for `quay.io` and the user explicitly wanted to create credentials for `quay.io/repo` (or `docker.io/ns` vs `docker.io/ns/repo`), what should happen?

That means another new function in c/image … and thinking what _exactly_ we should be checking for.

Reply:
Still outstanding, though giving up might be the right answer, short-term at least.

Reply:
… but we need to make sure that it is possible to change credentials from valid-for-registry-but-not-repo to valid-for-repo. Right now that seems possible but the user must explicitly specify `--username`; is that good enough?

Reply:
Should be try both, the full `key` as well as the `domain`?

Reply:
I think testing only `key` would make most sense. If we wanted to try `domain` as well to tell the user “you asked to login to `quay.io/mitr/test-server` but you already have credentials for `quay.io`”, wouldn’t we want to do the same check for the intermediate `quay.io/mitr` namespace as well?

That line of thought is not clearly nonsensical, but it’s also fiddly enough that doing the simplest thing and only checking `key` seems cleaner to me.